src/types/settings.ts

```diff
@@ -61,27 +79,119 @@ const smartCopilotSettingsSchema = z.object({
     model: '',
   }),
 
+  // System Prompt
   systemPrompt: z.string().catch(''),
-  ragOptions: z
-    .object({
-      chunkSize: z.number().catch(1000),
-      thresholdTokens: z.number().catch(8192),
-      minSimilarity: z.number().catch(0.0),
-      limit: z.number().catch(10),
-    })
-    .catch({
-      chunkSize: 1000,
-      thresholdTokens: 8192,
-      minSimilarity: 0.0,
-      limit: 10,
-    }),
+
+  // RAG Options
+  ragOptions: ragOptionsSchema.catch({
+    chunkSize: 1000,
+    thresholdTokens: 8192,
+    minSimilarity: 0.0,
+    limit: 10,
+  }),
 })
 
 export type SmartCopilotSettings = z.infer<typeof smartCopilotSettingsSchema>
 
+type Migration = {
+  fromVersion: string
+  toVersion: string
+  migrate: (data: Record<string, unknown>) => Record<string, unknown>
+}
+
+const MIGRATIONS: Migration[] = [
+  {
+    fromVersion: '0.0.0',
+    toVersion: '1.0.0',
+    migrate: (data) => {
+      const newData = { ...data }
+
+      if ('chatModel' in newData && typeof newData.chatModel === 'string') {
+        const CHAT_MODEL_MAP: Record<string, string> = {
+          'claude-3.5-sonnet-latest': 'anthropic/claude-3.5-sonnet-latest',
+          'gpt-4o': 'openai/gpt-4o',
+          'gpt-4o-mini': 'openai/gpt-4o-mini',
+          'llama-3.1-70b-versatile': 'groq/llama-3.1-70b-versatile',
+          'llama3.1:8b': 'ollama',
+        }
+        newData.chatModelId = CHAT_MODEL_MAP[newData.chatModel] ?? ''
+        delete newData.chatModel
+      }
+      if ('applyModel' in newData && typeof newData.applyModel === 'string') {
+        const APPLY_MODEL_MAP: Record<string, string> = {
+          'gpt-4o-mini': 'openai/gpt-4o-mini',
+          'llama-3.1-8b-instant': 'groq/llama-3.1-8b-instant',
+          'llama-3.1-70b-versatile': 'groq/llama-3.1-70b-versatile',
+          'llama3.1:8b': 'ollama',
+        }
+        newData.applyModelId = APPLY_MODEL_MAP[newData.applyModel] ?? ''
```

Comment:
It would be better to fall back to the default model if not found, instead of an empty string.